packages/snaps-execution-environments/src/common/BaseSnapExecutor.test.ts

```diff
@@ -587,6 +389,82 @@ describe('BaseSnapExecutor', () => {
     });
   });
 
+  it('allows direct access to snap public properties', async () => {
```

Comment:
(could) add a test proving prototype can't be accessed via `Object.getPrototypeOf` and `__proto__`; 
And then make sure it fails on the previous implementation ;)

Reply:
There is more proper test for testing security, inside `packages/snaps-execution-environments/src/common/BaseSnapExecutor.ava.test.ts` which is for AVA test runner only. I improved prototype checks there. The one for the Jest would not be checking the real (final) hardening since Jest does not work with SES properly. Some of these Jest tests are more for making peace with coverage results or checking basic functionality.

Reply:
To answer more properly on your request about `proving prototype can't be accessed via Object.getPrototypeOf`, I've added one more test that will use `Object.getPrototypeOf`, then compare that with `{}` using deepEqual from AVA. Because, `Object.getPrototypeOf` will still return an object, but that one is not the prototype of the `StreamProvider`.

Reply:
sounds good